File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.java
class Solution {
    public int minimumPushes(String word) {
      int[] fre =new int[26];
      for(char i:word.toCharArray()){
        fre[i-'a']++;
      }
      Arrays.sort(fre);
      int ans=0;
      int idx=0;
      for(int i=25;i>=0 &&fre[i]>0;i--){
         ans+=fre[i]*(idx/8+1);
         idx++;
      }
      return ans;
    }
}

File: 3276-minimum-number-of-pushes-to-type-word-ii/minimum-number-of-pushes-to-type-word-ii.py
class Solution(object):
    def minimumPushes(self, word):
        """
        :type word: str
        :rtype: int
        """
        fre=dict()
        for i in set(word):
            c=word.count(i)
            fre[i]=c
        ans=sorted(fre.items(),key=lambda x:x[1],reverse=True)   
        res=0
        for i in range(len(ans)):
            if i>23:
                res+=4*ans[i][1]
            elif i>15:
                res+=3*ans[i][1]
            elif i>7:
                res+=2*ans[i][1]
            else:
                res+=ans[i][1]
        return res                   